backbone/src/main/java/org/researchstack/backbone/step/active/recorder/DeviceMotionRecorder.java

```diff
@@ -70,7 +73,16 @@ public void start(Context context) {
     @Override
     protected void writeJsonData() {
         // Update the main json object
-        jsonObject.addProperty(TIMESTAMP_KEY, System.currentTimeMillis());
+        jsonObject.addProperty(TIMESTAMP_IN_SECONDS_KEY, System.currentTimeMillis() * 1e-3d);
+
+        double uptime;
```

Comment:
I remember from my meeting with Shannon that "uptime" was in fact elapsedRealtime from cpu boot time, but I thought we didn't need that because we had System.currentTimeMillis().  However, now that I think about it more, grabbing the system time at the point of writeJsonData() is a bug, and not accurate of when the sensor actually recorded the value.

Also, on further investigation, Android also uses "uptime" the same way iOS does for it's sensors.  See this post for more information https://stackoverflow.com/questions/5500765/accelerometer-sensorevent-timestamp

What we need to change this to is remove any timestamp setting in "writeJsonData" and move it to each recorder's onSensorChanged(SensorEvent sensorEvent), and grab the "uptime" value and set it there. 

Does this make sense?

Reply:
Yup, all makes sense, thanks.

Reply:
Seems then we should write the JSON on the sensor event and not use writeJsonData() and startRunnableForWritingJson() to control frequency.

PedometerRecorder uses the "manual frequency", but it's implementation is writing on onSensorEvent also, so that shouldn't cause an issue.

Reply:
We can do that for individual sensors just fine.  However, my goal was to group them together like in DeviceMotionSensor, and I didn't want to write all the data each time a single sensor changed it's value.  Does that make sense?  Any ideas on how to do that?